[2025-05-22T00:12:52.301+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-05-22T00:12:52.309+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: kafka_to_minio_to_spark.spark_etl scheduled__2025-05-21T16:00:00+00:00 [queued]>
[2025-05-22T00:12:52.315+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: kafka_to_minio_to_spark.spark_etl scheduled__2025-05-21T16:00:00+00:00 [queued]>
[2025-05-22T00:12:52.315+0000] {taskinstance.py:2865} INFO - Starting attempt 4 of 6
[2025-05-22T00:12:52.320+0000] {taskinstance.py:2888} INFO - Executing <Task(SSHOperator): spark_etl> on 2025-05-21 16:00:00+00:00
[2025-05-22T00:12:52.322+0000] {standard_task_runner.py:72} INFO - Started process 427 to run task
[2025-05-22T00:12:52.324+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'kafka_to_minio_to_spark', 'spark_etl', 'scheduled__2025-05-21T16:00:00+00:00', '--job-id', '197', '--raw', '--subdir', 'DAGS_FOLDER/testlog_ml.py', '--cfg-path', '/tmp/tmpq_ufc1rh']
[2025-05-22T00:12:52.325+0000] {standard_task_runner.py:105} INFO - Job 197: Subtask spark_etl
[2025-05-22T00:12:52.345+0000] {task_command.py:467} INFO - Running <TaskInstance: kafka_to_minio_to_spark.spark_etl scheduled__2025-05-21T16:00:00+00:00 [running]> on host f901d5ffba09
[2025-05-22T00:12:52.447+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='kafka_to_minio_to_spark' AIRFLOW_CTX_TASK_ID='spark_etl' AIRFLOW_CTX_EXECUTION_DATE='2025-05-21T16:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-05-21T16:00:00+00:00'
[2025-05-22T00:12:52.448+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-05-22T00:12:52.451+0000] {ssh.py:124} INFO - ssh_hook is not provided or invalid. Trying ssh_conn_id to create SSHHook.
[2025-05-22T00:12:52.455+0000] {base.py:84} INFO - Retrieving connection 'local_ssh'
[2025-05-22T00:12:52.455+0000] {ssh.py:309} WARNING - No Host Key Verification. This won't protect against Man-In-The-Middle attacks
[2025-05-22T00:12:52.506+0000] {transport.py:1923} INFO - Connected (version 2.0, client OpenSSH_9.8)
[2025-05-22T00:12:52.707+0000] {transport.py:1923} INFO - Authentication (password) successful!
[2025-05-22T00:12:52.708+0000] {ssh.py:493} INFO - Running command: sh -c "/Users/jeongmieun/.pyenv/versions/airminio/bin/python /Users/jeongmieun/test/docker_***/dags/testlog_ml_spark.py"
[2025-05-22T00:12:53.505+0000] {ssh.py:544} WARNING - 25/05/22 09:12:53 WARN Utils: Your hostname, jeongmieun-ui-MacBookAir.local resolves to a loopback address: 127.0.0.1; using 172.16.24.224 instead (on interface en0)
[2025-05-22T00:12:53.508+0000] {ssh.py:544} WARNING - 25/05/22 09:12:53 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2025-05-22T00:12:53.641+0000] {ssh.py:544} WARNING - 25/05/22 09:12:53 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-05-22T00:12:53.713+0000] {ssh.py:544} WARNING - Setting default log level to "WARN".
[2025-05-22T00:12:53.713+0000] {ssh.py:544} WARNING - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2025-05-22T00:12:54.796+0000] {ssh.py:544} WARNING - 25/05/22 09:12:54 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2025-05-22T00:12:57.379+0000] {ssh.py:544} WARNING - Traceback (most recent call last):
[2025-05-22T00:12:57.379+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/test/docker_***/dags/testlog_ml_spark.py", line 33, in <module>
[2025-05-22T00:12:57.380+0000] {ssh.py:544} WARNING -     .withColumn("like_score", when((col("event_type") == "like_click") & (col('liked') == 1, lit(3)))
[2025-05-22T00:12:57.380+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/pyspark/sql/column.py", line 163, in _
[2025-05-22T00:12:57.380+0000] {ssh.py:544} WARNING -     njc = getattr(self._jc, name)(jc)
[2025-05-22T00:12:57.381+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/py4j/java_gateway.py", line 1314, in __call__
[2025-05-22T00:12:57.381+0000] {ssh.py:544} WARNING -     args_command, temp_args = self._build_args(*args)
[2025-05-22T00:12:57.381+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/py4j/java_gateway.py", line 1277, in _build_args
[2025-05-22T00:12:57.381+0000] {ssh.py:544} WARNING -     (new_args, temp_args) = self._get_args(args)
[2025-05-22T00:12:57.381+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/py4j/java_gateway.py", line 1264, in _get_args
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -     temp_arg = converter.convert(arg, self.gateway_client)
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/py4j/java_collections.py", line 511, in convert
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -     java_list.add(element)
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/py4j/java_gateway.py", line 1314, in __call__
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -     args_command, temp_args = self._build_args(*args)
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/py4j/java_gateway.py", line 1277, in _build_args
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -     (new_args, temp_args) = self._get_args(args)
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/py4j/java_gateway.py", line 1264, in _get_args
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -     temp_arg = converter.convert(arg, self.gateway_client)
[2025-05-22T00:12:57.382+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/py4j/java_collections.py", line 510, in convert
[2025-05-22T00:12:57.383+0000] {ssh.py:544} WARNING -     for element in object:
[2025-05-22T00:12:57.383+0000] {ssh.py:544} WARNING -   File "/Users/jeongmieun/.pyenv/versions/airminio/lib/python3.10/site-packages/pyspark/sql/column.py", line 718, in __iter__
[2025-05-22T00:12:57.383+0000] {ssh.py:544} WARNING -     raise PySparkTypeError(
[2025-05-22T00:12:57.383+0000] {ssh.py:544} WARNING - pyspark.errors.exceptions.base.PySparkTypeError: [NOT_ITERABLE] Column is not iterable.
[2025-05-22T00:12:57.868+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 762, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/ssh/operators/ssh.py", line 190, in execute
    result = self.run_ssh_client_command(ssh_client, self.command, context=context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/ssh/operators/ssh.py", line 178, in run_ssh_client_command
    self.raise_for_status(exit_status, agg_stderr, context=context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/ssh/operators/ssh.py", line 172, in raise_for_status
    raise AirflowException(f"SSH operator error: exit status = {exit_status}")
airflow.exceptions.AirflowException: SSH operator error: exit status = 1
[2025-05-22T00:12:57.873+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=kafka_to_minio_to_spark, task_id=spark_etl, run_id=scheduled__2025-05-21T16:00:00+00:00, execution_date=20250521T160000, start_date=20250522T001252, end_date=20250522T001257
[2025-05-22T00:12:57.881+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-05-22T00:12:57.881+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 197 for task spark_etl (SSH operator error: exit status = 1; 427)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 762, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/ssh/operators/ssh.py", line 190, in execute
    result = self.run_ssh_client_command(ssh_client, self.command, context=context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/ssh/operators/ssh.py", line 178, in run_ssh_client_command
    self.raise_for_status(exit_status, agg_stderr, context=context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/providers/ssh/operators/ssh.py", line 172, in raise_for_status
    raise AirflowException(f"SSH operator error: exit status = {exit_status}")
airflow.exceptions.AirflowException: SSH operator error: exit status = 1
[2025-05-22T00:12:57.913+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-05-22T00:12:57.922+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-05-22T00:12:57.922+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
